### PR TITLE
Clientside toggle for first person embarks/executions

### PR DESCRIPTION
**Pull request by Gityeeter**
-Additional convars for enabling first person embarks/executions. 
-Also added Titanfall 1 style executions as a option from classicTitanexecution mod.

https://github.com/R2Northstar/NorthstarMods/assets/94848105/201c4242-6e7b-4e59-8895-0079fe1cd1f3

### CODE REVIEW

**Comment by ASpoonPlaysGames** [comment]
So when you say "Clientside toggle" you just mean that you use convars instead of the playlistvar, and add them to modsettings?

**Comment by Gityeeter** [comment]
> So when you say "Clientside toggle" you just mean that you use convars instead of the playlistvar, and add them to modsettings?

That is correct. AFAIK you can't change playlistvar from clientside. Convars can be changed even during the middle of a game.

File: Northstar.Custom/mod.json [hunk 1]
```diff
@@ -24,6 +24,18 @@
...
+			"Name": "execution_style",
+			"DefaultValue": "0"
+		},
+		{
+			"Name": "first_person_embark",
+			"DefaultValue": "0"
+		},
+		{
+			"Name": "execution_style_pilot",
+			"DefaultValue": "0"
```
Comment (F1F7Y) [inline Northstar.Custom/mod.json:38]: If you're going to rely on modsettings add the archive flag so they save

File: Northstar.Custom/mod.json [hunk 2]
```diff
@@ -426,6 +438,13 @@
 		{
 			"Path":  "sh_northstar_http_requests.gnut",
 			"RunOn": "CLIENT || SERVER || UI"
+		},
+		{
+			"Path": "embark_settings.nut",
+			"RunOn": "CLIENT || UI",
```
Comment (F1F7Y) [inline Northstar.Custom/mod.json:444]: I don't think you need `CLIENT`

File: Northstar.Custom/mod/scripts/vscripts/embark_settings.nut [hunk 3]
```diff
@@ -0,0 +1,13 @@
+untyped
+global function embarkSettings
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/embark_settings.nut:2]: Should be named `EmbarkSettings_Init` to follow the style of other callbacks

File: Northstar.Custom/mod/scripts/vscripts/embark_settings.nut [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+
+void function embarkSettings(){
+    #if UI
+    AddModTitle("Embark/Execution Settings")
+    AddModCategory("General")
+    AddConVarSettingEnum("execution_style", "Titan Execution", ["Default", "First Persons", "Titanfall 1 Style"])
+    AddConVarSettingEnum("execution_style_pilot", "Pilot Execution", ["Default", "First Person"])
+    AddConVarSettingEnum("first_person_embark", "Embark", ["Third Person", "First Person"])
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/embark_settings.nut:11]: Mod settings has been merged and now has differently named functions, use those

File: Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_human.gnut [hunk 5]
```diff
@@ -259,16 +259,17 @@ void function MeleeThread_PilotVsEnemy_Attacker( SyncedMelee action, entity atta
 	if ( isAttackerRef )
 	{
 		#if MP
-		if ( GetCurrentPlaylistVarInt( "fp_embark_enabled", 0 ) == 1 )
+
+		if (GetConVarInt("execution_style_pilot") == 1)
 			FirstPersonSequenceForce1P( attackerSequence, attacker )
 		#endif
-			
+
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_human.gnut:266]: Unnecessary whitespace changes in this file

File: Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_titan.gnut [hunk 6]
```diff
@@ -242,6 +242,39 @@ bool function MeleeThread_TitanVsTitan_Internal( SyncedMelee action, entity atta
 
 void functionref( SyncedMelee action, entity attacker, entity target ) function GetTitanSyncedMeleeFunc( entity attacker, entity target )
 {
+	print(GetConVarInt("execution_style"))
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_titan.gnut:245]: No need to print

File: Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_titan.gnut [hunk 7]
```diff
@@ -242,6 +242,39 @@ bool function MeleeThread_TitanVsTitan_Internal( SyncedMelee action, entity atta
...
+	if ( GetConVarInt("execution_style") == 2 )
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/melee/_melee_synced_titan.gnut:246]: Should be an enum to make the code more readable imo

File: Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut [hunk 8]
```diff
@@ -2231,4 +2269,4 @@ void function SetEmbarkDebugPrint( bool value )
 {
 	file.embarkDebugPrint = value
 }
-#endif
\ No newline at end of file
+#endif
```
Comment (F1F7Y) [inline Northstar.Custom/mod/scripts/vscripts/titan/sh_titan_embark.gnut:2272]: again

**Review by F1F7Y — CHANGES_REQUESTED** [review]
Just code formatting, havent tested or read the logic fully yet

**Comment by F1F7Y** [comment]
> -Additional convars for enabling first person embarks/executions.
> -Also added Titanfall 1 style executions as a option from classicTitanexecution mod.

Should be separate PRs

**Pull request closed without merging**